formula/src/test/java/com/instacart/formula/subjects/StartStopFormula.kt

```diff
@@ -40,13 +42,16 @@ class StartStopFormula(runtime: TestableRuntime) : Formula<Unit, State, Output>
             },
             output = Output(
                 state = state.count,
-                startListening = context.onEvent {
-                    transition(state.copy(listenForEvents = true))
-                },
-                stopListening = context.onEvent {
-                    transition(state.copy(listenForEvents = false))
-                }
+                // We need to specify keys since `UpdateListenFlag` type is used two times.
+                startListening = context.onEvent("start", UpdateListenFlag(listen = true)),
+                stopListening = context.onEvent("stop", UpdateListenFlag(listen = false)),
             )
         )
     }
+
+    private class UpdateListenFlag(val listen: Boolean): Transition<State, Unit> {
+        override fun TransitionContext<State>.toResult(event: Unit): Transition.Result<State> {
+            return transition(state.copy(listenForEvents = listen))
```

Comment:
The `state` is provided by the `TransitionContext`. I'm thinking it would be nice to also provide the `Input`, however, this would most likely require adding the `Input` type to `FormulaContext`.

Reply:
To support `Input` type and maybe to improve concepts, maybe `FormulaContext` should be renamed to `EvaluationContext<Input, State>` and we could simplify logic to 
```kotlin
interface Formula<Input, State, Output> {
    fun EvaluationContext<Input, Output>.evaluate(): Evaluation<Output> 
}
```

There is still a bit to consider here, but would be an interesting approach where you would get `input` and `state` from the `EvaluationContext`.

Reply:
I like EvaluationContext, it better describes what this does 👍 

Due to how it is used on Android "Context" kind of sounds like some god object which isn't tied to this specific evaluation

I don't know if this is actually tripping people discovering formula though ?